src/components/Input/Input.scss

```diff
@@ -7,11 +7,17 @@
         @each $context in (brand, primary, accent, info, good, warning, bad) {
             &_#{ $context } {
                 &:focus {
-                    border-color: var(--color-#{ $context });
-                    box-shadow: 0 0 0 var(--outline-size) var(--color-#{ $context }-25);
+                    border-color: var(--color-#{$context});
+                    box-shadow: 0 0 0 var(--outline-size) var(--color-#{$context}-25);
                 }
             }
         }
+        @each $context in (warning, bad) {
+            &_#{ $context } {
+                border-color: var(--color-#{$context});
+                box-shadow: 0 0 0 var(--outline-size) var(--color-#{$context}-25);
+            }
+        }
```

Comment:
Also, few points below are out of scope of the story, but I'd like to know your opinion on that.

1. Currently, context mixes too much stuff together: validation colors, primary/secondary colors and other palette values.
2. Because of point 1) naming is confusing and not consistent (neutral - adj, brand - noun, primary - adj, accent - noun, info - noun, good - adj, warning - verb/noun, bad - adj)
3. For different components different colors can mean different things (or can have no meaning at all).
4. Not every component needs (makes sense) with every context values.

Some examples:

- `Input`. Only few context values make sense from my pov: `default`, `bad`, `good`, `warning`.
- `Button`.  Sensible values:`primary/accent` and secondary.
- `Alert`, the same as for input..
and so on..

So if you share the same feeling, I'd like to give it some more thoughts and come up with a better scheme for this.

Reply:
to be honest, I don't get confused by it. I think this naming is used in other libraries as well, but I don't know of examples from the top of my head. Also, to your original comment, I thought that this change is what we decided in the meetings - change the way these contexts behave rather then adding a new prop.

Reply:
> I thought that this change is what we decided in the meetings - change the way these contexts behave rather then adding a new prop.

Yes, but in this PR discussed change implemented only for `bad` and `warning` context values. For all the other values behavior is still the same as it was. I think it also should be changed. Otherwise it's confusing.

Reply:
applying the same changes to other contexts, such as brand or primary will not make visual sense.

Reply:
Well, that's my point from the previous comment (see point 3 from https://github.com/textkernel/oneui/pull/225#discussion_r371148395)

It doesn't make sense now as well if you think about it.

So I'd rather limit context values for Input to only those which make sense, or at least make the behavior consistent.

Reply:
I'm fine either way. @carlobernardini ?

Reply:
as per f2f discussion all contexts but `good, warning, bad` will be removed from Input